Bulking requests to Slack's conversation listing

When archiving multiple channels at the end of the session, we run into 
```
Error listing conversations: The request to the Slack API failed. (url: https://www.slack.com/api/conversations.list)
The server responded with: {'ok': False, 'error': 'ratelimited'}
Error archiving channel: The request to the Slack API failed. (url: https://www.slack.com/api/conversations.archive)
The server responded with: {'ok': False, 'error': 'invalid_arguments', 'response_metadata': {'messages': ['[ERROR] missing required field: channel']}}
``` 

That is likely because we do multiple calls to `conversations.list`, one per channel, to retrieve the channel ID to archive. To avoid this, we should bulk the calls to extract the multiple channel IDs and then iterate to archive the conversations.

--- a/interfaces/slack/SlackInterface.py
+++ b/interfaces/slack/SlackInterface.py
@@ -43,9 +43,11 @@ class SlackInterface:
 
             # if we find the channel, it ends here
             for channel in result['channels']:
-                if channel['name'] == name:
-                    self.channel_dict[name] = channel['id']
-                    return channel['id']
+                # populate channel_dict for all channels retrieved
+                self.channel_dict[channel['name']] = channel['id']
+
+            if name in self.channel_dict:
+                return self.channel_dict[name]
 
             # if it was not found, check next cursor
             # at the end, next_cursor will be an empty string

